nilearn/plotting/img_plotting.py

```diff
@@ -37,6 +39,50 @@
 # Core, usage-agnostic functions
 
 
+def check_threshold(threshold, data, percentile_calculate, name=None):
```

Comment:
I think that there is a misunderstanding on this function. You seem to randomly add stuff to address comments without converging toward an acceptable solution. What you must do is see the logic in common between the two functions, see which functional part of the code can be put in common, think about the specifications (or write them down) and then code it.